diff --git a/third_party/amd/python/examples/gluon/moe_gfx1250.py b/third_party/amd/python/examples/gluon/moe_gfx1250.py
index 2155d97611..5c842984c8 100644
--- a/third_party/amd/python/examples/gluon/moe_gfx1250.py
+++ b/third_party/amd/python/examples/gluon/moe_gfx1250.py
@@ -2197,8 +2197,10 @@ def matmul(a, b, bias, a_ragged_metadata: RaggedTensorMetadata | None = None,
     else:
         batch_size = 1
 
+    intermediate_out_dtype = precision_config.intermediate_out_dtype or torch.float32
     opt_flags = make_opt_flags(out_dtype, a.dtype, b.dtype, precision_config, batch_size, M, N, b.shape[-2],
-                               a_ragged_metadata, False, False, False, False, False, block_k=block_k)
+                               a_ragged_metadata, False, False, False, False, False, block_k=block_k,
+                               intermediate_out_dtype=intermediate_out_dtype)
 
     opt_flags.block_m = block_m
     opt_flags.block_n = block_n
@@ -2212,7 +2214,7 @@ def matmul(a, b, bias, a_ragged_metadata: RaggedTensorMetadata | None = None,
 
     # init output
     allocation = init_allocation(a, b, precision_config, fused_activation, gather_indx, scatter_indx, batch_size, 1,
-                                 opt_flags)
+                                 opt_flags, intermediate_out_dtype)
     memory = apply_allocation(allocation, None)
     # (1, batch_dim, M, N // fused_activation.specs)
     out_matmul = memory["scratchpad"].get("matmul", memory["output"])
